@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -485,11 +482,19 @@ export class TextElementsRenderer {
      * `true` if any resource used by any `FontCatalog` is still loading.
      */
     get loading(): boolean {
-        let isLoading = this.m_fontCatalogLoader.loading;
-        for (const textRenderer of this.m_textRenderers) {
-            isLoading = isLoading || textRenderer.textCanvas.fontCatalog.isLoading;
+        return this.m_fontCatalogLoader.loading || this.m_glyphLoadingCount > 0;
+    }
+
+    /**
+     * Waits till all pending resources from any `FontCatalog` are loaded.
+     */
+    async waitLoaded(): Promise<void> {
+        this.waitInitialized();
+
+        if (this.m_loadPromise !== undefined) {
+            await this.m_loadPromise;
         }
-        return isLoading;
+        return;
```

Comment:
Here I would also return Promise<bool>, thus, the code would look like this:
`
this.waitInitialized();

if (this.m_loadPromise === undefined) {
    return false
}	 
await this.m_loadPromise;
return true;
`
this will be more consistent with another `waitInitilized()' method.

Reply:
done